### PR TITLE
[ashell] Adding code to free cb memory and change cleanup order

### PR DESCRIPTION
**Pull request by brianjjones**
This adds a method that ensures all callbacks are freed.  It
also changes the order of some of the cleanup functions to
ensure stability when starting / stopping apps using ashell.

Signed-off-by: Brian J Jones <brian.j.jones@intel.com>

### CODE REVIEW

File: src/zjs_callbacks.c [hunk 1]
```diff
@@ -326,6 +326,15 @@ void zjs_remove_callback(zjs_callback_id id)
     }
 }
 
+void zjs_remove_all_callback()
```
Comment (grgustaf) [inline src/zjs_callbacks.c]: maybe callbacks?

**Comment by grgustaf** [comment]
Other than that, LGTM.

File: src/zjs_callbacks.c [hunk 2]
```diff
@@ -326,6 +326,15 @@ void zjs_remove_callback(zjs_callback_id id)
     }
 }
 
+void zjs_remove_all_callbacks()
+{
+   for (int i = 0; i < cb_size; i++) {
+        if (cb_map[i]) {
+            zjs_remove_callback(i);
```
Comment (jimmy-huang) [inline src/zjs_callbacks.c]: Does this code clean up the ringbuffer? maybe @jprestwo can take a look here.

Comment (brianjjones) [inline src/zjs_callbacks.c]: Probably not, but I don't think ring buffer was the issue I was seeing.  cb_map was the item making zjs_malloc calls.  Not saying ring buffer couldn't also be an issue however.

**Comment by poussa** [comment]
I was testing this. It seems to fix the hang issue (#745) or at least improve it significantly. However, there seems to be on new issue, which I guess is in this PR.

Using the sample app from the (#745), on first run you get one callback when you press the button. When you upload the app again (2nd run), you get two callbacks when you press the button. And so on. If you upload the app 10 times, you get 10 callbacks per button press.

But it did not hang!

**Comment by brianjjones** [comment]
@poussa Strange, especially since (in theory) I'm deleting all the callbacks :)

Thanks for trying it out, I'll try and reproduce and fix it so that its a complete fix.

**Pull request merged**